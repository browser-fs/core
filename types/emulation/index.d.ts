export * from './async';
export * from './sync';
export * as promises from './promises';
export * as constants from './constants';
export * from './streams';
export * from './dir';
export { mountMapping, mounts, mount, umount, _toUnixTimestamp } from './shared';
export { Stats, BigIntStats, StatsFs } from '../stats';
