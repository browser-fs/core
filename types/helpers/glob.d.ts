/**
 * Synchronous, recursive function to list files matching a glob pattern in a directory and its subdirectories using minimatch.
 * @param dirPath Directory to search within
 * @param pattern Glob pattern to match filenames (e.g., '*.txt')
 * @returns Array of matching filenames, including path relative to the initial directory
 */
export declare function globSync(dirPath: string, pattern: string): string[];
