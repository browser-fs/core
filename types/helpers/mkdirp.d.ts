/**
 * Synchronously creates a directory and all necessary parent directories.
 * @param dirPath The path of the directory to create.
 */
export declare function mkdirpSync(dirPath: string): void;
