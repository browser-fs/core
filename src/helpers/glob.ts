import { join } from '../emulation/path';
import { minimatch } from 'minimatch';
import { readdirSync } from '../process/os_sync'

/**
 * Synchronous, recursive function to list files matching a glob pattern in a directory and its subdirectories using minimatch.
 * @param dirPath Directory to search within
 * @param pattern Glob pattern to match filenames (e.g., '*.txt')
 * @returns Array of matching filenames, including path relative to the initial directory
 */
export function globSync(dirPath: string, pattern: string): string[] {
  let results: string[] = [];

  function recurse(currentPath) {
    const entries = readdirSync(currentPath, { withFileTypes: true });

    for (let entry of entries) {
      const fullPath = join(currentPath, entry.name);
      if (entry.isDirectory()) {
        recurse(fullPath); // Recursively search in subdirectory
      } else if (minimatch(entry.name, pattern)) {
        results.push(fullPath); // Add file to results if it matches the pattern
      }
    }
  }

  try {
    recurse(dirPath); // Start the recursion from the initial directory path
    return results;
  } catch (err) {
    console.error('Error:', err);
    return []; // Return an empty array in case of an error
  }
}
