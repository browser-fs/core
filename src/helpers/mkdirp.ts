import { join } from '../emulation/path';
import { existsSync, mkdirSync } from '../process/os_sync'

/**
 * Synchronously creates a directory and all necessary parent directories.
 * @param dirPath The path of the directory to create.
 */
export function mkdirpSync(dirPath: string) {
  const parentPath = join(dirPath, '..');  // Get the parent directory

  if (!existsSync(parentPath)) {
    mkdirpSync(parentPath);  // Recursively ensure parent directories exist
  }

  if (!existsSync(dirPath)) {
    mkdirSync(dirPath);  // Create the directory if it does not exist
  }
}

