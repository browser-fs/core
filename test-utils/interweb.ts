import { configure, InMemory } from '../src';

import { fs } from '../src';
import { join, relative } from 'path';
import { statSync, readFileSync, readdirSync } from 'fs';

export const fixturesDir = '__fixtures__/interweb';

export function setup(_p: string = fixturesDir) {
  const p = relative(fixturesDir, _p) || '/';
  const stats = statSync(_p);

  if (!stats.isDirectory()) {
    fs.writeFileSync(p, readFileSync(_p));
    return;
  }

  if (p != '/') {
    fs.mkdirSync(p);
  }
  for (const file of readdirSync(_p)) {
    setup(join(_p, file));
  }
}

export { fs };
