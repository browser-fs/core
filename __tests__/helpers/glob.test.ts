import { join } from '../../src/emulation/path';
import { minimatch } from 'minimatch';
import { globSync } from '../../src/helpers/glob';
import { fs, setup } from '../../test-utils/interweb';

beforeAll(setup);

it('globSync', () => {
  const files = globSync('./pyramation', '*.txt');
  expect(files).toMatchSnapshot();
});