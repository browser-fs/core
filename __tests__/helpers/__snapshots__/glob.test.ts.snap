// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globSync 1`] = `
[
  "pyramation/deployments/deployment-config.txt",
]
`;
